crates/cast/src/helpers/scarb_utils.rs

```diff
@@ -42,14 +117,13 @@ pub fn get_profile<'a>(tool_sncast: &'a Value, profile: &Option<String>) -> Resu
     }
 }
 
-pub fn get_property<'a, T>(tool: &'a Value, field: &str) -> T
+pub fn get_property<T>(tool: &Value, field: &str) -> T
 where
-    T: From<&'a str> + Default,
+    T: PropertyFromCastConfig + Default,
 {
     tool.get(field)
-        .and_then(Value::as_str)
-        .map(T::from)
-        .unwrap_or_default()
+        .and_then(T::from_toml_value)
```

Comment:
I wonder how this will distinguish between u8 and u16 and apply the correct default value?

Reply:
it infers the type, and the type must have implemented PropertyFromCastConfig for this to work, hence the lines 79-97